src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -146,6 +167,136 @@ namespace OpenMS
+    params.setValue("deisotoping:annotate_charge", "false", "Annotate the charge to the peaks");
+  }
+
+  void TargetedSpectraExtractor::annotateSpectra(
+      const std::vector<MSSpectrum>& spectra,
+      const FeatureMap& ms1_features,
+      FeatureMap& ms2_features,
+      std::vector<MSSpectrum>& annotated_spectra) const
+  {
+    annotated_spectra.clear();
+    for (const auto& spectrum : spectra)
+    {
+      if (spectrum.getMSLevel() == 1)
+      {
+        continue; // we want to annotate MS2 spectra only
+      }
+
+      const double spectrum_rt = spectrum.getRT();
+      const double rt_left_lim = spectrum_rt - rt_window_ / 2.0;
+      const double rt_right_lim = spectrum_rt + rt_window_ / 2.0;
+      const std::vector<Precursor>& precursors = spectrum.getPrecursors();
+      if (precursors.empty())
+      {
+        OPENMS_LOG_WARN << "annotateSpectra(): No precursor MZ found. Setting spectrum_mz to 0." << std::endl;
+      }
+      const double spectrum_mz = precursors.empty() ? 0.0 : precursors.front().getMZ();
+      const double mz_tolerance = mz_unit_is_Da_ ? mz_tolerance_ : mz_tolerance_ / 1e6;
+
+      // When spectrum_mz is 0, the mz check on transitions is inhibited
+      const double mz_left_lim = spectrum_mz ? spectrum_mz - mz_tolerance : std::numeric_limits<double>::min();
+      const double mz_right_lim = spectrum_mz ? spectrum_mz + mz_tolerance : std::numeric_limits<double>::max();
+
+      for (const auto& feature : ms1_features)
+      {
+        for (const auto& subordinate : feature.getSubordinates())
+        {
+          const auto& peptide_ref = subordinate.getMetaValue("PeptideRef");
+          const double target_mz = subordinate.getMZ();
+          const double target_rt = subordinate.getRT();
+          if (target_rt >= rt_left_lim && target_rt <= rt_right_lim)
+          {
+            std::cout << "annotateSpectra(): " << peptide_ref << "]";
+            std::cout << " (target_rt: " << target_rt << ") (target_mz: " << target_mz << ")" << std::endl
+                      << std::endl;
+            MSSpectrum annotated_spectrum = spectrum;
+            annotated_spectrum.setName(peptide_ref);
+            annotated_spectra.push_back(annotated_spectrum);
+            // fill the ms2 features map
+            Feature ms2_feature;
+            ms2_feature.setRT(spectrum_rt);
+            ms2_feature.setMZ(spectrum_mz);
+            ms2_feature.setIntensity(subordinate.getIntensity());
+            ms2_feature.setMetaValue("transition_name", peptide_ref);
+            ms2_features.push_back(ms2_feature);
+          }
+        }
+      }
+    }
+  }
+
+  void TargetedSpectraExtractor::searchSpectrum(
+      OpenMS::FeatureMap& feat_map,
+      OpenMS::FeatureMap& feat_map_output) const
+  {
+    OpenMS::AccurateMassSearchEngine ams;
+    OpenMS::MzTab output;
+    ams.init();
+    ams.run(feat_map, output);
```

Comment:
When AMSE is initialized inside the `searchSpectrum()`, the database and adduct list can not be updated, or? Which means the algorithm, only works with the current HMDB (4.0) database. There is no way to use a custom database or to reduce the adduct space. 

Is that the desired, or should there be a possibility to change the default?